### PR TITLE
Add support for YAML config files

### PR DESCRIPTION
**Pull request by austintraver**
* Increase the clarity of the usage documentation
* Add the `requirements.txt` dependency file
* Include the PyYAML package

### CODE REVIEW

File: README.md
Comment (saadmk11) [inline README.md:23]: These are and/or option, So, you can have both or just one of them (Commit or Comment)

File: README.md
Comment (saadmk11) [inline README.md:51]: Reference `(#config-file-usage)` does not match with the new header of that section

File: README.md
Comment (saadmk11) [inline README.md:54]: Reference `(#config-file-usage-optional)` does not match with the new header of that section

File: README.md
Comment (saadmk11) [inline README.md:92]: I think we should keep the ` (Optional):`

File: scripts/changelog-ci.py
Comment (saadmk11) [inline scripts/changelog-ci.py:15]:
```suggestion
                       r"0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?"
```

File: scripts/changelog-ci.py
Comment (saadmk11) [inline scripts/changelog-ci.py:72]: We are not using annotations anywhere rights now

File: scripts/changelog-ci.py
Comment (saadmk11) [inline scripts/changelog-ci.py:76]: Don't we need to close the file?

File: scripts/changelog-ci.py
Comment (saadmk11) [inline scripts/changelog-ci.py:2]: We should add this with `requests` as it's not a built-in library.

**Review by saadmk11 — CHANGES_REQUESTED** [review]
Thank you for your contribution here :100: 

I have not tested this locally, but the changes look good :100:, I have left some comments regarding minor changes to the documentation. :)

It would be great if you could include a sample `YAML` config in the documentation :) https://github.com/saadmk11/changelog-ci#example-config-file

File: README.md
Comment (austintraver) [reply README.md:23]: Good catch, will resolve this in the next commit.

File: README.md
Comment (austintraver) [reply README.md:92]: I agree that configurations themselves should be optional, but I think users will find this page easier to read if we stick to the common convention. I made this change after consulting `man(1) man`, what I consider to be a somewhat canonical source of truth for writing documentation:

>  A manual page consists of several sections.
> 
> Conventional  section  names  include **NAME**, **SYNOPSIS**, **CONFIGURATION**, **DESCRIPTION**, **OPTIONS** ... **NOTES**, **BUGS**, **EXAMPLE**, **AUTHORS**, and **SEE ALSO**.

Perhaps a compromise: we emphasize that configurations are ***optional*** directly below the `Configuration` heading

File: scripts/changelog-ci.py
Comment (austintraver) [reply scripts/changelog-ci.py:72]: Very well. I will revert this in the next commit.

File: README.md
Comment (saadmk11) [reply README.md:92]: I still feel like it's more straightforward to put it in the header. It will be easy to find and create less confusion :)

Comment (austintraver) [reply README.md:92]: I see where you're coming from. Fixed in https://github.com/saadmk11/changelog-ci/pull/46/commits/a00cd665a8dbfea2bd12149ee7c227181d66caf9

File: scripts/changelog-ci.py
Comment (saadmk11) [reply scripts/changelog-ci.py:2]: I am referring to [PEP 8](https://www.python.org/dev/peps/pep-0008/#imports)

> Imports should be grouped in the following order:
> 
> Standard library imports.
> Related third party imports.
> Local application/library specific imports.
> You should put a blank line between each group of imports.

File: scripts/changelog-ci.py
Comment (saadmk11) [reply scripts/changelog-ci.py:76]: Yes, the garbage collector will eventually close the file, but Isn't it good practice to always explicitly close the file without relying on the garbage collector?

Comment (austintraver) [reply scripts/changelog-ci.py:76]: I agree it is good practice to do so, but my response was answering your first question:

> Don't we need to close the file?

Comment (saadmk11) [reply scripts/changelog-ci.py:76]: Oh, Alright 👍 The PR is almost ready to get merged 💯, I think we should close the opened file here 🙂

File: README.md
Comment (saadmk11) [reply README.md:92]: Looks great, Thanks 💯

File: scripts/changelog-ci.py
Comment (austintraver) [reply scripts/changelog-ci.py]: Ah, thank you for explaining! I understand now. I have pushed an updated commit. https://github.com/saadmk11/changelog-ci/pull/46/commits/92be906b9bdd74d149243d7fb9fe33cc7751707e resolves all issues that violate compliance with PEP 8.

File: scripts/changelog-ci.py
Comment (austintraver) [reply scripts/changelog-ci.py:76]: Go for it! You have write access to this branch

**Comment by saadmk11** [comment]
I don't think we need docker-compose for this action as it only runs one container

File: scripts/changelog-ci.py
Comment (saadmk11) [reply scripts/changelog-ci.py:76]: Sure, I'm merging this to another branch before merging to master as I need to test the YAML config as well, thanks for your great work on this :)

**Review by saadmk11 — APPROVED** [review]
🎉🎉

**Comment by austintraver** [comment]
> I don't think we need docker-compose for this action as it only runs one container

Good point. Compose is truly, as you said, better suited for multi-container applications.

**Pull request merged**
